File: Go/14_test.go
package main

import (
	"fmt"
	"testing"
)

func TestLongestCommonPrefix(t *testing.T) {
	tt := []struct {
		strs []string
		want string
	}{
		{strs: []string{"flower", "flow", "flight"}, want: "fl"},
		{strs: []string{"dog", "racecar", "car"}, want: ""},
	}

	for i, test := range tt {
		t.Run(fmt.Sprintf("case %d", i), func(t *testing.T) {
			got := longestCommonPrefix(test.strs)
			want := test.want
			if got != want {
				t.Errorf("Got %s but want %s", got, want)
			}
		})
	}
}

File: C++/tests/14_test.cpp
#include <gtest/gtest.h>
#include <solution_tmpl.h>
using namespace std;

class LongestCommonPrefixTest : public ::testing::Test {
protected:
  Solution solution;
};

TEST_F(LongestCommonPrefixTest, CommonPrefixExists) {
  vector<string> strs = {"flower", "flow", "flight"};
  string want = "fl";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, NoCommonPrefix) {
  vector<string> strs = {"dog", "racecar", "car"};
  string want = "";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, SingleString) {
  vector<string> strs = {"alone"};
  string want = "alone";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, EmptyInput) {
  vector<string> strs = {};
  string want = "";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, NoCommonPrefixAtAll) {
  vector<string> strs = {"abc", "def", "ghi"};
  string want = "";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, AllStringsIdentical) {
  vector<string> strs = {"same", "same", "same"};
  string want = "same";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}

TEST_F(LongestCommonPrefixTest, PrefixIsTheShortestString) {
  vector<string> strs = {"ab", "abc", "abcd"};
  string want = "ab";

  string actual = solution.longestCommonPrefix(strs);
  EXPECT_EQ(actual, want);
}
